Description: Remove early termination behavior during sensor detection
 Exit the application when all sensors on the system are determined to be unsupported
 rather than exiting immediately after encountering the first unsupported sensor.
Author: Nanley Chery <nanleychery@gmail.com>
---
This patch header follows DEP-3: http://dep.debian.net/deps/dep3/
--- a/src/gui.c
+++ b/src/gui.c
@@ -438,6 +438,7 @@
 
     int chipnum = 0;
     const sensors_chip_name *name = NULL;
+    int supported_sensor_found = FALSE;
 
     gtk_init( &argc, &argv );
 
@@ -507,15 +508,17 @@
 #ifdef DEBUG_XSENSORS
             printf( "Adding tab for %s\n", name->prefix );
 #endif
-            if ( ( head = add_sensor_tab( notebook, name ) ) == NULL )
-                return FAILURE;
-            
-            update_sensor_data( head );
-            g_signal_connect( G_OBJECT (mainwindow), "realize",
-                              G_CALLBACK (start_timer), head );
+            if ( ( head = add_sensor_tab( notebook, name ) ) != NULL ) {
+                supported_sensor_found = update_sensor_data( head );
+                g_signal_connect( G_OBJECT (mainwindow), "realize",
+                                  G_CALLBACK (start_timer), head );
+            }
         }
     }
     
+    if ( supported_sensor_found != TRUE )
+        return FAILURE;
+
     /* Setup the main components. */
     gtk_widget_show( mainwindow );
 
